src/node_lttng.cc

```diff
@@ -0,0 +1,264 @@
+      ->ToInteger(env->isolate())->Value();
+
+#define SLURP_OBJECT(obj, member, valp) \
+  if (!(obj)->IsObject()) { \
+    return env->ThrowError( \
+      "expected object for " #obj " to contain object member " #member); \
+  } \
+  *valp = Local<Object>::Cast(obj->Get(OneByteString(env->isolate(), #member)));
+
+#define SLURP_CONNECTION(arg, conn) \
+  if (!(arg)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg " to be a connection object"); \
+  } \
+  node_lttng_connection_t conn; \
+  Local<Object> _##conn = Local<Object>::Cast(arg); \
+  Local<Value> _handle = \
+      (_##conn)->Get(FIXED_ONE_BYTE_STRING(env->isolate(), "_handle")); \
+  if (_handle->IsObject()) { \
+    SLURP_INT(_handle.As<Object>(), fd, &conn.fd); \
+  } else { \
+    conn.fd = -1; \
+  } \
+  SLURP_STRING(_##conn, remoteAddress, &conn.remote); \
+  SLURP_INT(_##conn, remotePort, &conn.port); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered);
+
+#define SLURP_CONNECTION_HTTP_CLIENT(arg, conn) \
+  if (!(arg)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg " to be a connection object"); \
+  } \
+  node_lttng_connection_t conn; \
+  Local<Object> _##conn = Local<Object>::Cast(arg); \
+  SLURP_INT(_##conn, fd, &conn.fd); \
+  SLURP_STRING(_##conn, host, &conn.remote); \
+  SLURP_INT(_##conn, port, &conn.port); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered);
+
+#define SLURP_CONNECTION_HTTP_CLIENT_RESPONSE(arg0, arg1, conn) \
+  if (!(arg0)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg0 " to be a connection object"); \
+  } \
+  if (!(arg1)->IsObject()) { \
+    return env->ThrowError( \
+      "expected argument " #arg1 " to be a connection object"); \
+  } \
+  node_lttng_connection_t conn; \
+  Local<Object> _##conn = Local<Object>::Cast(arg0); \
+  SLURP_INT(_##conn, fd, &conn.fd); \
+  SLURP_INT(_##conn, bufferSize, &conn.buffered); \
+  _##conn = Local<Object>::Cast(arg1); \
+  SLURP_STRING(_##conn, host, &conn.remote); \
+  SLURP_INT(_##conn, port, &conn.port);
+
+
+void LTTNG_NET_SERVER_CONNECTION(const FunctionCallbackInfo<Value>& args) {
+  if (!NODE_NET_SERVER_CONNECTION_ENABLED())
+    return;
+  Environment* env = Environment::GetCurrent(args);
+  SLURP_CONNECTION(args[0], conn);
+  NODE_NET_SERVER_CONNECTION(&conn, conn.remote, conn.port, conn.fd);
+}
+
+
+void LTTNG_NET_STREAM_END(const FunctionCallbackInfo<Value>& args) {
+  if (!NODE_NET_STREAM_END_ENABLED())
+    return;
+  Environment* env = Environment::GetCurrent(args);
+  SLURP_CONNECTION(args[0], conn);
+  NODE_NET_STREAM_END(&conn, conn.remote, conn.port, conn.fd);
+}
+
+
+void LTTNG_HTTP_SERVER_REQUEST(const FunctionCallbackInfo<Value>& args) {
+  node_lttng_http_server_request_t req;
+
+  if (!NODE_HTTP_SERVER_REQUEST_ENABLED())
+    return;
+
+  Environment* env = Environment::GetCurrent(args);
+  HandleScope scope(env->isolate());
+  Local<Object> arg0 = Local<Object>::Cast(args[0]);
```

Comment:
Caveat emptor: Cast() does the wrong thing when someone does `this.connection = null`.  There should be a `if (!args[0]->IsObject()) return` guard a few lines up.  You probably want to do it after the NODE_HTTP_SERVER_REQUEST_ENABLED check because that check is presumably cheaper than calling `args[0]->IsObject()`.

Side note: you don't need the HandleScope.  JS -> C++ callbacks have an implicit HandleScope.